modules/oe_subscriptions_anonymous/tests/src/FunctionalJavascript/SubscribeModalFormTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_subscriptions_anonymous\Functional;
+
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\Tests\flag\Traits\FlagCreateTrait;
+
+/**
+ * Modal form test.
+ */
+class SubscribeModalFormTest extends WebDriverTestBase {
+
+  use FlagCreateTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'node',
+    'flag',
+    'oe_subscriptions',
+    'oe_subscriptions_anonymous',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * Tests that subscribe link open a modal.
+   */
+  public function testModalForm(): void {
+    // Create an article content type.
+    $this->drupalCreateContentType([
+      'type' => 'article',
+      'name' => 'Article',
+    ]);
+    // Create a flag.
+    $flag = $this->createFlagFromArray([
+      'id' => 'subscribe_article',
+      'short_text' => 'Subscribe article',
+      'entity_type' => 'node',
+      'bundles' => ['article'],
+    ]);
+    // Create the node.
+    $node = $this->drupalCreateNode([
+      'type' => 'article',
+      'status' => 1,
+    ]);
+    $node->save();
+
+    $page = $this->getSession()->getPage();
+    $assert_session = $this->assertSession();
+    // Using ui-dialog parent of modal-form,
+    // Submit buttons and close in siblings.
+    $modal_selector = '.ui-dialog';
+
+    // Got to node page with subscription.
+    $this->drupalGet($node->toUrl());
+    // Click subscribe link.
+    $link_text = $flag->getShortText('flag');
```

Comment:
You set a defined value on top, you shouldn't use the flag method here. You are expecting that text on top to be used for the flag. So you can safely use "Subscribe article" (even better if we change it to a realistic "Subscribe to this article").

Reply:
Done